### PR TITLE
task-9

### PR DESCRIPTION
**Pull request by svetamint**
Task-9: Add parameters to the query
In this task, we need to extend the capabilities of the existing endpoint /meteors.

We will pass query parameters in the query to control the logic on the server.

Query params:

-date helps to set the period during which we are interested in information on meteorites

-count returns only the number of meteorites that have been seen.

-were-dangerous-meteors returns true or false whether they have been seen among meteorites potentially dangerous to Earth.

P.S. consider the possibility of transferring several parameters at the same time

Acceptance criteria:

Add date as a parameter to the query
Add count as a parameter to the query
Add were-dangerous-meteors as a parameter to the request
Consider the possibility of passing several, all or none of the parameters

### CODE REVIEW

File: task9/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
+import {config} from "./config/config.js";
+import meteorRouter from './delivery/meteorController.js'
+
+const app = express();
+
+app.use(meteorRouter)
+
```
Comment (Mikhail1992) [inline task9/index.js:8]: add `errorHandler` please somewhere there

here you can find a little bit more information https://medium.com/@mohsinogen/simplified-guide-to-setting-up-a-global-error-handler-in-express-js-daf8dd640b69

File: task9/delivery/meteorController.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const meteorRouter = express.Router()
+
+meteorRouter.get('/meteors', async (request, response) => {
+    try {
+        const { date, count, wereDangerousMeteors } = request.query;
+        const { startDate, endDate } = getDateRange(date)
+        const meteorDto = await getMeteorDto(startDate, endDate, wereDangerousMeteors, count);
+        return response.json(meteorDto);
+    } catch (error) {
+        return response.status(500).json({ error: `Failed to get meteors info due to: ${error.message}` });
```
Comment (Mikhail1992) [inline task9/delivery/meteorController.js:14]:
```
next(error)
```
Error - can be a custom object
Example
```js
class Exception extends Error {
  constructor(code, massage) {
    this.code = code;
    this.massage = massage;
  }
}

...

next(new Exception(500, 'Error message')
```

File: task9/use_cases/meteorMapper.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { getMeteorData } from '../repository/meteorsClient.js'
+
+export const getMeteorDto = async (startDate, endDate, wereDangerousMeteors, count) => {
+    try {
```
Comment (Mikhail1992) [inline task9/use_cases/meteorMapper.js:4]: You can remove try..catch here and leave it only in the controller

File: task9/repository/meteorsClient.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import axios from 'axios';
+
+import { config } from "../config/config.js";
+
+export const getMeteorData = async(startDate, endDate) => {
+    try {
```
Comment (Mikhail1992) [inline task9/repository/meteorsClient.js:6]: You can remove try..catch here and leave it only in the controller

File: task9/repository/meteorsClient.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+        const nasaUrl = `${config.nasaMeteorUrl}&start_date=${startDate}&end_date=${endDate}`;
+        const response = await axios.get(nasaUrl);
```
Comment (Mikhail1992) [inline task9/repository/meteorsClient.js:8]: it is possible to use some alternative syntax
```
axios.get(config.nasaMeteorUrl, {
    params: {
      start_date: startDate,
      end_date: endDate
    }
  })
```

File: task9/delivery/meteorController.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import { getMeteorDto } from '../use_cases/meteorMapper.js'
+import { config } from "../config/config.js";
+
+const meteorRouter = express.Router()
+
+meteorRouter.get('/meteors', async (request, response) => {
+    try {
+        const { date, count, wereDangerousMeteors } = request.query;
+        const { startDate, endDate } = getDateRange(date)
+        const meteorDto = await getMeteorDto(startDate, endDate, wereDangerousMeteors, count);
```
Comment (Mikhail1992) [inline task9/delivery/meteorController.js:11]: wereDangerousMeteors and count boolean data types. Convert please strings to boolean type

Comment (svetamint) [reply task9/delivery/meteorController.js]: done

File: task9/delivery/meteorController.js (hunk 2 above)
Comment (svetamint) [reply task9/delivery/meteorController.js]: done

File: task9/index.js (hunk 1 above)
Comment (svetamint) [reply task9/index.js:9]: done

File: task9/repository/meteorsClient.js (hunk 4 above)
Comment (svetamint) [reply task9/repository/meteorsClient.js]: done

File: task9/repository/meteorsClient.js (hunk 5 above)
Comment (svetamint) [reply task9/repository/meteorsClient.js]: done

File: task9/use_cases/meteorMapper.js (hunk 3 above)
Comment (svetamint) [reply task9/use_cases/meteorMapper.js]: done

File: task9/exception/errorHandler.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const errorHandler = (error, request, response, next) => {
+    response.status(500).json(error.message);
```
Comment (Mikhail1992) [inline task9/exception/errorHandler.js:2]:
```
response.status(error.code || 500).json(error.message);
```

File: task9/index.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import {config} from "./config/config.js";
+import meteorRouter from './delivery/meteorController.js'
+import errorHandler from './exception/errorHandler.js';
+
+const app = express();
+
+app.use(meteorRouter, errorHandler)
```
Comment (Mikhail1992) [inline task9/index.js:8]: this is not a mistake but better to split routers and error handlers

```js
app.use(meteorRouter)
app.use(errorHandler)
```

Comment (svetamint) [reply task9/index.js:8]: fixed

File: task9/index.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (Mikhail1992) [inline task9/index.js:8]: you forgot to remove errorHandler from `app.use(meteorRouter, errorHandler)` :)

Comment (svetamint) [reply task9/index.js]: oops, fixed

**Review by Mikhail1992 — APPROVED** [review]
Looks good

**Pull request merged**